### PR TITLE
Hot fix setup config json on web

### PR DESCRIPTION
**Pull request by nqhhdev**
### Root cause:

- When we use `AppConfig.homeserver` in `AutoHomeserverPickerController` on web platform however `AppConfig.homeserver` not sure to set up new data from `config.json`. 
→ Can't get login `[Matrix] Found no well known information - XMLHttpRequest error.`

### Solution

- Use `completer` make sure to set up new data from `config.json`. 

### Case failed

https://github.com/linagora/twake-on-matrix/assets/99852347/0666274e-4a97-4357-af72-7192c3ff8cd0

### Case success

https://github.com/linagora/twake-on-matrix/assets/99852347/176b4a06-d6b2-4e8f-89d9-9c011193ee23

### CODE REVIEW

File: lib/presentation/mixins/init_config_mixin.dart [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    } else {
+      AppConfig.initConfigCompleter.complete(false);
+    }
+  }
+
+  Future<void> initConfigMobile() async {
+    try {
+      AppConfig.loadEnvironment();
+    } catch (e) {
+      Logs().v('[ConfigLoader] config.json not found', e);
```
Comment (hoangdat) [inline lib/presentation/mixins/init_config_mixin.dart:38]: change message error for the case of mobile

File: lib/pages/auto_homeserver_picker/auto_homeserver_picker.dart [hunk 2]
```diff
@@ -91,15 +94,15 @@ class AutoHomeserverPickerController extends State<AutoHomeserverPicker>
...
+      autoHomeserverPickerUIState.value = AutoHomeServerPickerFailureState();
       Logs().d(
         "AutoHomeserverPickerController: _autoCheckHomeserver: Error: $e",
       );
     }
   }
 
   void retryCheckHomeserver() {
-    showButtonRetryNotifier.toggle();
+    autoHomeserverPickerUIState.value = AutoHomeServerPickerLoadingState();
```
Comment (hoangdat) [inline lib/pages/auto_homeserver_picker/auto_homeserver_picker.dart:105]: please change state to Loading also in `initState`

File: lib/config/app_config.dart [hunk 3]
```diff
@@ -9,6 +10,12 @@ import 'package:matrix/matrix.dart';
 abstract class AppConfig {
   static ResponsiveUtils responsive = getIt.get<ResponsiveUtils>();
 
+  static Completer<bool> initConfigCompleter = Completer<bool>();
+
+  static int retryCompleterCount = 0;
+
+  static bool get maxRetryCompleter => retryCompleterCount == 3;
```
Comment (sherlockvn) [inline lib/config/app_config.dart:17]: `
static bool get hasReachedMaxRetries => retryCompleterCount == 3;
`

File: lib/pages/auto_homeserver_picker/auto_homeserver_picker.dart [hunk 4]
```diff
@@ -21,12 +22,14 @@ class AutoHomeserverPicker extends StatefulWidget {
...
 
 class AutoHomeserverPickerController extends State<AutoHomeserverPicker>
-    with ConnectPageMixin {
+    with ConnectPageMixin, InitConfigMixin {
   static const Duration autoHomeserverPickerTimeout = Duration(seconds: 30);
 
   static const _saasPlatform = 'saas';
 
-  final showButtonRetryNotifier = ValueNotifier<bool>(false);
+  final autoHomeserverPickerUIState = ValueNotifier<AutoHomeServerPickerState>(
```
Comment (sherlockvn) [inline lib/pages/auto_homeserver_picker/auto_homeserver_picker.dart:30]: do we need to dispose this one somewhere ?

**Review by hoangdat — APPROVED** [review]

**Review by sherlockvn — APPROVED** [review]

**Pull request merged**